src/utils/clmfates_interfaceMod.F90

```diff
@@ -1095,30 +1138,45 @@ subroutine wrap_update_hlmfates_dyn(this, nc, bounds_clump,      &
           ! protext it here with a lower bound of 0.0_r8.
 
           patch%wt_ed(col%patchi(c)) = max(0.0_r8, &
+
                1.0_r8-sum(this%fates(nc)%bc_out(s)%canopy_fraction_pa(1:npatch)))
 
-          if(sum(this%fates(nc)%bc_out(s)%canopy_fraction_pa(1:npatch))>1.0_r8)then
-             write(iulog,*)'Projected Canopy Area of all FATES patches'
-             write(iulog,*)'cannot exceed 1.0'
-             !end_run()
-          end if
+          patch%sp_pftorder_index(col%patchi(c)) = 0 !bg is the 0th patch in the SP FATES structure
 
-          do ifp = 1, npatch
+          areacheck = patch%wt_ed(col%patchi(c)) ! this is where we start the areachecking
+
+          do ifp = 1, this%fates(nc)%sites(s)%youngest_patch%patchno
+             ! for the vegetated patches
 
              p = ifp+col%patchi(c)
 
              ! bc_out(s)%canopy_fraction_pa(ifp) is the area fraction
-             ! the site's total ground area that is occupied by the 
-             ! area footprint of the current patch's vegetation canopy 
+             ! the site's total ground area that is occupied by the
+             ! area footprint of the current patch's vegetation canopy
 
              patch%is_veg(p) = .true.
              patch%wt_ed(p)  = this%fates(nc)%bc_out(s)%canopy_fraction_pa(ifp)
+             areacheck = areacheck + patch%wt_ed(p)
+
              elai(p) = this%fates(nc)%bc_out(s)%elai_pa(ifp)
-             tlai(p) = this%fates(nc)%bc_out(s)%tlai_pa(ifp)
              esai(p) = this%fates(nc)%bc_out(s)%esai_pa(ifp)
-             tsai(p) = this%fates(nc)%bc_out(s)%tsai_pa(ifp)
              hbot(p) = this%fates(nc)%bc_out(s)%hbot_pa(ifp)
-             htop(p) = this%fates(nc)%bc_out(s)%htop_pa(ifp)
+
+             if(use_fates_sp)then
+               canopystate_inst%tlai_hist_patch = this%fates(nc)%bc_out(s)%tlai_pa(ifp)
+               canopystate_inst%tsai_hist_patch = this%fates(nc)%bc_out(s)%tsai_pa(ifp)
```

Comment:
```suggestion
               canopystate_inst%tsai_hist_patch(p) = this%fates(nc)%bc_out(s)%tsai_pa(ifp)
```

Reply:
Corrected

Reply:
I'm kind of surprised fortran does not generate an error for this. I feel like one should have to at least use the brackets a colon for "all indices" to make this a viable statement.  Maybe there is a compiler directive we should start enforcing that imposes this behavior.

Reply:
agreed that that is worth investigating, but I suspect it is valid fortran unfortunately, and so not "catchable" via a compiler directive?

Reply:
Yes, it's valid FORTRAN, because it means you are setting the entire array to the single value. There are times that we do this on purpose -- although I agree that you should do it explicitly with a array(:) = constant form. It is a common problem to look for that causes issues. I can think of several times this has happened. I don't know of any compiler that has a directive to catch this. So I don't think there's a way to catch this automatically.

Reply:
Thanks @ekluzek